### PR TITLE
CodeMirror preview option

### PR DESCRIPTION
**Pull request by Hamms**
[Jira](https://codedotorg.atlassian.net/secure/RapidBoard.jspa?rapidView=28&projectKey=FND&view=detail&selectedIssue=FND-158)

Update initializeCodeMirror to:
1. take an options object rather than an ordered set of optional arguments
2. accept a "preview" option to control the preview area (still defaults to trying to infer the preview area from the id of the target if option is unspecified)

Also update the authored hints editor to _use_ said preview option, rather than implementing a preview area manually.

### CODE REVIEW

File: apps/src/code-studio/initializeCodeMirror.js [hunk 1]
```diff
@@ -38,17 +38,17 @@ const INVALID_COLOR = '#d00';
...
  * @param {!string|!Element} target - element or id of element to replace.
  * @param {!string} mode - editor syntax mode
- * @param {function} [callback] - onChange callback for editor
- * @param {booblen} [attachments] - whether to enable attachment uploading in
- *        this editor.
+ * @param {Object} options - misc optional arguments
+ * @param {function} [options.callback] - onChange callback for editor
+ * @param {boolean} [options.attachments] - whether to enable attachment
+ *        uploading in this editor.
+ * @param {???} [options.onUpdateLinting]
```
Comment (islemaster) [inline apps/src/code-studio/initializeCodeMirror.js]:
```suggestion
 * @param {function} [options.onUpdateLinting]
```

File: apps/src/code-studio/initializeCodeMirror.js [hunk 2]
```diff
@@ -38,17 +38,17 @@ const INVALID_COLOR = '#d00';
...
+ * @param {!string!Element} [options.preview] - element or id of element to
```
Comment (islemaster) [inline apps/src/code-studio/initializeCodeMirror.js]:
```suggestion
 * @param {(string|Element)} [options.preview] - element or id of element to
```
I think this is the JSDoc you're looking for. ([source](http://usejsdoc.org/tags-type.html))

**Review by islemaster — APPROVED** [review]
Nice!

File: apps/src/code-studio/initializeCodeMirror.js (hunk 2 above)
Comment (Hamms) [reply apps/src/code-studio/initializeCodeMirror.js]: d'oh, yes it is. Thanks!

**Pull request merged**